diff --git a/src/sage/dynamics/complex_dynamics/mandel_julia.py b/src/sage/dynamics/complex_dynamics/mandel_julia.py
index f26602e908..5c6e662800 100644
--- a/src/sage/dynamics/complex_dynamics/mandel_julia.py
+++ b/src/sage/dynamics/complex_dynamics/mandel_julia.py
@@ -130,13 +130,13 @@ def mandelbrot_plot(x_center=-1.0,
     Notebook, set ``interact`` to ``True``::
 
         sage: mandelbrot_plot(interact=True)
-        interactive(children=(FloatSlider(value=-1.0, description=u'Real center'...
+        interactive(children=(...FloatSlider(value=-1.0, description=u'Real center'...
 
     ::
 
         sage: mandelbrot_plot(interact=True, x_center=-0.75, y_center=0.25,
         ....:     image_width=1/2, number_of_colors=75)
-        interactive(children=(FloatSlider(value=-0.75, description=u'Real center'...
+        interactive(children=(...FloatSlider(value=-0.75, description=u'Real center'...
     """
     base_color = Color(base_color)
 
@@ -415,7 +415,7 @@ def julia_plot(c=-1,
     set ``interact`` to ``True``::
 
         sage: julia_plot(interact=True)
-        interactive(children=(FloatSlider(value=-1.0, description=u'Real c'...
+        interactive(children=(...FloatSlider(value=-1.0, description=u'Real c'...
 
     To return the Julia set of a random `c` value with (formal) cycle structure
     `(2,3)`, set ``period = [2,3]``::
